service/history/execution/context.go

```diff
@@ -45,10 +45,29 @@ const (
 	defaultRemoteCallTimeout = 30 * time.Second
 )
 
-var (
-	// ErrConflict is exported temporarily for integration test
-	ErrConflict = errors.New("conditional update failed")
-)
+type conflictError struct {
+	Cause error
+}
+
+func (e *conflictError) Error() string {
+	return fmt.Sprintf("conditional update failed: %s", e.Cause.Error())
```

Comment:
```suggestion
	return fmt.Sprintf("conditional update failed: %v", e.Cause)
```

Since I haven't actually looked into it in detail before: this should probably be `... %v", e.Cause`.

Because:
`fmt.Errorf` does this: https://github.com/golang/go/blob/master/src/fmt/errors.go#L17-L30
and, with the `Error()` impl which just returns that message, is basically "eagerly create the full message".  
TIL.  
That `wrapErrs` flag later reaches here in the fmt printer: https://github.com/golang/go/blob/master/src/fmt/print.go#L574-L587
which just swaps it with `%v` for stringifying, and holds onto the error value (which is ignored by Errorf).  
TIL 2.

That's... simple and fairly backwards-compatible, but yow.  Worse than I expected.  Potentially unnecessarily costly and destroys a lot of information for no good reason (doesn't check for or support `%+v` or `%#v` because it can't).